hudi-hadoop-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -454,11 +457,11 @@ public Pair<ByteArrayOutputStream, Object> serializeRecordsToLogBlock(HoodieStor
     config.setValue(PARQUET_MAX_FILE_SIZE.key(), String.valueOf(1024 * 1024 * 1024));
 
     HoodieFileWriter parquetWriter = HoodieFileWriterFactory.getFileWriter(
-        HoodieFileFormat.PARQUET, outputStream, storage, config, writerSchema, recordType);
+        HoodieFileFormat.PARQUET, outputStream, storage, config, HoodieSchema.fromAvroSchema(writerSchema), recordType);
     while (recordItr.hasNext()) {
       HoodieRecord record = recordItr.next();
       String recordKey = record.getRecordKey(readerSchema, keyFieldName);
-      parquetWriter.write(recordKey, record, writerSchema);
+      parquetWriter.write(recordKey, record, HoodieSchema.fromAvroSchema(writerSchema));
```

Comment:
Similarly here, we can call `HoodieSchema.fromAvroSchema` once above at line 460 and reuse.